// heracles/hls_plugin_share.js
import { callNative } from './hls_plugin_base.js'
import { saveShareAppInstallStatus, getShareAppInstallStatus } from './hls_app_env_storage.js'

const SHARE_MODULE = "share"

/**
 * 分享插件
 */
class Share {
		
	/**
	 * 获取当前第三方客户端安装情况
	 */
	getAppInstallStatus() {
		/**
		 * 安装状态
		 * {
			 "qq": true,
			 "wechat": true,
			 "weibo": true
		 }
		 */
		return new Promise(function(resolve, reject) {
			callNative(SHARE_MODULE,'getAppInstallStatus',{},"获取分享App的安装状态").then(res => {
				saveShareAppInstallStatus(res)
				if(typeof resolve == 'function') {
					resolve(res)
				}
			}).catch(error => {
				if(typeof reject == 'function') {
					reject(error)
				}
			})
		})
	}
	
	/**
	 * 分享到微信好友
	 * @param {Object} data
	 * title
	 * description 
	 * imageUrl 
	 * url
	 * applet 小程序页面分享参数
	 * {
		 path: "index/index",
		 params: {}
	 }
	 * @param {Object} callback
	 */
	shareToWeChat(data) {
	   return callNative(SHARE_MODULE,'shareToWeChat',data,"分享给微信好友")
	}
	
	/**
	 * 分享到微信朋友圈
	 * @param {Object} data
	 */
	shareToWeChatTimeline(data) {
	   return callNative(SHARE_MODULE,'shareToWeChatTimeline',data,"分享到微信朋友圈")
	}
	
	/**
	 * 分享到微博
	 * @param {Object} data
	 */
	shareToWeibo(data) {
	   return callNative(SHARE_MODULE,'shareToWeibo',data,"分享到微博")
	}
	
	/**
	 * 分享到QQ
	 * @param {Object} data
	 */
	shareToQQ(data) {
	   return callNative(SHARE_MODULE,'shareToQQ',data,"分享到QQ")
	}
	
	/**
	 * 拉起分享面板
	 * @param {Object} data
	 */
	showSharePanel(data) {
		return callNative(SHARE_MODULE,'showSharePanel',data,"拉起分享面板")
	}
	
	/**
	 * 设置右上角分享按钮
	 * @param {Object} data
	 */
	setShareInfo(data) {
		return callNative(SHARE_MODULE,'setShareInfo',data,"设置右上角分享按钮")
	}
}

export default Share;
