// heracles/hls_app_env_storage.js

const HLS_SHARE_APP_INSTALL_STATUS = "HLS_SHARE_APP_INSTALL_STATUS"

const HLS_LOGIN_USER_INFO = "HLS_LOGIN_USER_INFO"

const HLS_LOGIN_USER_TOKEN = "HLS_LOGIN_USER_TOKEN"

const HLS_HOST_APP_CONFIG = "HLS_HOST_APP_CONFIG"

/**
 * 通过key通用保存数据到持久存储中
 * @param {string} key - 存储的键名
 * @param {Object} data - 存储的数据
 */
function saveToStorage(key, data) {
	try {
		uni.setStorageSync(key, data)
		console.log(key+"保存数据成功!")
	} catch (exception) {
		console.log(key+"保存数据失败!")
	}
}

/**
 * @param {string} key - 存储的键名
 */
function getByKey(key) {
	try {
		console.log(key+"获取数据成功!")
		return uni.getStorageSync(key)
	} catch (exception) {
		console.log(key+"获取数据失败!")
	}
}

/**
 * @param {string} key - 存储的键名
 */
function removeByKey(key) {
	try {
		uni.removeStorageSync(key)
		console.log(key+"删除数据成功!")
	} catch (exception) {
		console.log(key+"删除数据失败!")
	}
}

/**
 * @param {Object} data - 保存第三方App(微信、微博、QQ)的安装状态
 */
export function saveShareAppInstallStatus(data) {
	saveToStorage(HLS_SHARE_APP_INSTALL_STATUS, data)
}

/**
 * 获取第三方App的安装状态
 */
export function getShareAppInstallStatus() {
	getByKey(HLS_SHARE_APP_INSTALL_STATUS)
}

/**
 * @param {string} token - 用户登录Token
 */
export function saveLoginUserToken(token) {
	saveToStorage(HLS_LOGIN_USER_TOKEN, token)
}

/**
 * 获取用户登录的Token
 */
export function getLoginUserToken() {
	return getByKey(HLS_LOGIN_USER_TOKEN)
}

/**
 * 删除用户的登录Token
 */
export function removeLoginUserToken() {
	removeByKey(HLS_LOGIN_USER_TOKEN)
}

/**
 * 获取登录用户的基本信息，头像，昵称，性别
 */
export function getLoginUserInfo() {
	return getByKey(HLS_LOGIN_USER_INFO)
}

/**
 * @param {Object} data - 登录用户的基本信息
 */
export function saveLoginUserInfo(data) {
	saveToStorage(HLS_LOGIN_USER_INFO, data)
}

/**
 * 删除登录用户的基本信息
 */
export function removeLoginUserInfo() {
	removeByKey(HLS_LOGIN_USER_INFO)
}

/**
 * 保存宿主App的环境配置信息
 * @param {Object} config
 */
export function saveHostAppEnvConfig(config)
{
	saveToStorage(HLS_HOST_APP_CONFIG, config)
}

/**
 * 获取宿主App的环境配置信息
 */
export function getHostAppEnvConfig() 
{
	return getByKey(HLS_HOST_APP_CONFIG)
}
